barretenberg/cpp/src/barretenberg/dsl/acir_format/ivc_recursion_constraint.test.cpp

```diff
@@ -52,17 +48,15 @@ class IvcRecursionConstraintTest : public ::testing::Test {
      * @param num_public_inputs Number of public inputs to be extracted from the proof
      * @return RecursionConstraint
      */
-    static RecursionConstraint create_recursion_constraint(const VerifierInputs& input,
-                                                           SlabVector<FF>& witness,
-                                                           const size_t num_public_inputs)
+    static RecursionConstraint create_recursion_constraint(const VerifierInputs& input, SlabVector<FF>& witness)
     {
         // Assemble simple vectors of witnesses for vkey and proof
         std::vector<FF> key_witnesses = input.honk_verification_key->to_field_elements();
         std::vector<FF> proof_witnesses = input.proof; // proof contains the public inputs at this stage
 
         // Construct witness indices for each component in the constraint; populate the witness array
-        auto [key_indices, proof_indices, public_inputs_indices] =
-            ProofSurgeon::populate_recursion_witness_data(witness, proof_witnesses, key_witnesses, num_public_inputs);
+        auto [key_indices, proof_indices, public_inputs_indices] = ProofSurgeon::populate_recursion_witness_data(
+            witness, proof_witnesses, key_witnesses, /*num_public_inputs_to_extract=*/0);
```

Comment:
its defaulted to zero here because in actual Aztec there will not be NO public inputs aside from those that are fixed, e.g. aggregation object, databus, etc. Everything that used to use public inputs now uses the databus.